lua/kickstart/plugins/debug.lua

```diff
@@ -64,9 +64,67 @@ return {
     {
       '<leader>B',
       function()
-        require('dap').set_breakpoint(vim.fn.input 'Breakpoint condition: ')
+        require 'dap.protocol'
+        local dap = require 'dap'
+        -- Search for an existing breakpoint on this line in this buffer
+        ---@return dap.SourceBreakpoint bp that was either found, or an empty placeholder
+        local function find_bp()
+          local buf_bps = require('dap.breakpoints').get(vim.fn.bufnr())[vim.fn.bufnr()]
+          ---@type dap.SourceBreakpoint
+          local bp = { condition = '', logMessage = '', hitCondition = '', line = vim.fn.line '.' }
+          for _, candidate in ipairs(buf_bps) do
+            if candidate.line and candidate.line == vim.fn.line '.' then
+              bp = candidate
+              break
+            end
+          end
+          return bp
+        end
```

Comment:
This simplifies the code a bit
```suggestion
        local dap = require 'dap'

        -- Search for an existing breakpoint on this line in this buffer
        ---@return dap.SourceBreakpoint bp that was either found, or an empty placeholder
        local function find_bp()
          local buf_bps = require('dap.breakpoints').get(vim.fn.bufnr())[vim.fn.bufnr()]
          ---@type dap.SourceBreakpoint
          for _, candidate in ipairs(buf_bps) do
            if candidate.line and candidate.line == vim.fn.line '.' then
              return candidate
            end
          end

          return { condition = '', logMessage = '', hitCondition = '', line = vim.fn.line '.' }
        end

```